### DISCUSSION TITLE
Can we have a single scheduler with multiple databases?

### DISCUSSION THREAD

**Discussion by viveknuna** [quartznet/quartznet#2244] [Q&A]
I am using the Quartz 3.5.0 version in .Net 6 application. Its multitenant application. I use a dictionary with tenantId as the key and scheduler as the value. So based on the request for a given tenant, my code picks the respective schedule and works as expected.

Now I would like to have only a single schedule but multiple tenant databases ( which have quartz-related tables).

So I create an instance of schedule and at runtime based on the tenantId value I create a new schedule and assign it to the same schedule instance.

It's working fine. but there is an issue, jobs/triggers are created in the same database ( the one which is used for the very first request( only, ideally, it should point to the respective database.

**Comment by viveknuna** [comment]
@lahma  do you have any inputs on this?

**Reply by viveknuna** [reply]
@jafin do you have inputs on this?

**Reply by jafin** [reply]
@viveknuna IIRC IScheduler is still a singleton, therefore to achieve this you may need multiple schedulers, but given the singleton constraint it may be a challenge.  Take a look at https://github.com/quartznet/quartznet/issues/1453 , perhaps further investigation and if you are able to submit a PR may advance this?
